primitives/runtime/src/generic/digest.rs

```diff
@@ -173,6 +178,10 @@ pub enum DigestItemRef<'a, Hash: 'a> {
 	/// Digest item that contains signal from changes tries manager to the
 	/// native code.
 	ChangesTrieSignal(&'a ChangesTrieSignal),
+
+	/// Reference to `DigestItem::MerkleMountainRangeRoot`.
+	MerkleMountainRangeRoot(&'a Hash),
```

Comment:
Why isn't this done via `Other` case? You can create your own extension trait in your own code.
To handle the conversion like so:

```rust
trait AsMerkleMountainRangeRoot<Hash> {
 fn as_merkle_mountain_range_root(&self) -> Option<&Hash>;
}

impl<'a, Hash: 'a> AsMerkleMountainRangeRoot<Hash> for DigestItemRef<'a, Hash> {
  fn as_merkle_mountain_range_root(&self) -> Option<&Hash> {
     match *self {
			DigestItemRef::Other(ref data) => Hash::from(&data).ok(),
			_ => None,
		}
  }
}
```

Reply:
It makes sense.

I was thinking that if we use ```Other```, the data may conflict with digest item having the same struct which could be introduced in future modules. 

I think can resolve this by using a specific prefix in ```Other``` data.

Thanks, I'm closing this PR.

Reply:
AFAICT `Other` is meant to be exactly this - a non-conflicting way to add your own thing to `Digest`. Unless someone corrects me ;)
It is indeed to only one element though, but internally you can then encode multiple different variants.
Perhaps we are lacking some documentation here to make it clear what's the intention of that variant.